lorien/InfiniteCanvas/InfiniteCanvas.gd

```diff
@@ -345,4 +346,13 @@ func _undo_delete_stroke(stroke: BrushStroke) -> void:
 
 # -------------------------------------------------------------------------------------------------
 func _on_window_resized() -> void:
-	_viewport.size = get_viewport_rect().size
+	# Multiplying by scale needed to fix canvas after _set_scale() workaround
+	_viewport.size = get_viewport_rect().size * _scale
+
+# -------------------------------------------------------------------------------------------------
+func _set_scale(scale: float) -> void:
+	_scale = scale
+	_grid._grid_size = Config.DEFAULT_GRID_SIZE * _scale
+	_grid._on_viewport_size_changed()
```

Comment:
Instead of accessing the private grid variables directly it would be better to add a new public method to `InfiniteCanvasGrid` (something like `set_size`) which would then execute these two lines internally.